shm/src/buy.cpp

```diff
@@ -0,0 +1,61 @@
+#include <iostream>
+
+#include "buy.hpp"
+#include "map.hpp"
+
+Buy::Buy(Map* map) : map_(map) {}
+
+void Buy::displayStore(std::shared_ptr<Store> store) {
+    std::cout << "\n" << *store.get() << "\n";
+}
+
+void Buy::execute(Player* player) {
+    while (true) {
+        auto store = map_->getCurrentPosition()->getStore();
+        displayStore(store);
+
+        size_t position{};
+        size_t quantity{};
+
+        std::cout
+            << "Please enter the position and amount of cargo you are interested in";
+        std::cin >> position >> quantity;
```

Comment:
Validate user input. I will not point this problem further. So remember fix this in whole cases :)